pkg/controllers/util/planner/planner.go

```diff
@@ -215,10 +215,18 @@ func getDesiredPlan(
 	plan := make(map[string]int64, len(preferences))
 	overflow := make(map[string]int64, len(preferences))
 
+	// This algorithm does not support negative estimatedCapacity values.
+	santizedEstimatedCapacity := map[string]int64{}
```

Comment:
This probably needs to be at the top of `Plan`, or outside it.

Reply:
agreed. will move it